packages/components/src/components/combobox/component.tsx

```diff
@@ -0,0 +1,528 @@
+import type {
+	HideErrorPropType,
+	IdPropType,
+	InputTypeOnDefault,
+	KoliBriHorizontalIcons,
+	LabelWithExpertSlotPropType,
+	MsgPropType,
+	NamePropType,
+	ComboboxAPI,
+	ComboboxStates,
+	Stringified,
+	SyncValueBySelectorPropType,
+	TooltipAlignPropType,
+} from '../../schema';
+import { Component, Element, h, Host, Method, Prop, State, Watch, Fragment, Listen } from '@stencil/core';
+
+import { nonce } from '../../utils/dev.utils';
+import { stopPropagation, tryToDispatchKoliBriEvent } from '../../utils/events';
+import { ComboboxController } from './controller';
+
+import type { JSX } from '@stencil/core';
+import { KolIconTag, KolInputTag } from '../../core/component-names';
+import { showExpertSlot } from '../../schema';
+import { InternalUnderlinedAccessKey } from '../span/InternalUnderlinedAccessKey';
+import { getRenderStates } from '../input/controller';
+
+/**
+ * @slot - Die Beschriftung des Eingabefeldes.
+ */
+@Component({
+	tag: 'kol-combobox',
+	styleUrls: {
+		default: './style.scss',
+	},
+	shadow: true,
+})
+export class KolCombobox implements ComboboxAPI {
+	@Element() private readonly host?: HTMLKolComboboxElement;
+	private ref?: HTMLSelectElement;
+
+	@Method()
+	// eslint-disable-next-line @typescript-eslint/require-await
+	public async getValue(): Promise<string | undefined> {
+		return this._value;
+	}
+
+	private toggleListbox = () => {
+		this._isOpen = !this._isOpen;
+	};
+
+	private selectOption(option: string) {
+		this.state._value = option;
+		this.toggleListbox();
+	}
+	private onInput(event: Event) {
+		const target = event.target as HTMLInputElement;
+		this.state._value = target.value;
+		this.setFilteredOptionsByQuery(this.state._value);
+	}
+
+	private setFilteredOptionsByQuery(query: string) {
+		if (query.trim() === '') {
+			this._filtredOptions = [...this.state._options];
+		} else {
+			this._filtredOptions = this.state._options.filter((option: string) => {
+				return option.toLowerCase().includes(query.toLowerCase());
+			});
+		}
+	}
+
+	private _focusedOptionIndex: number = -1;
+
+	private moveFokus(delta: number) {
+		if (this._filtredOptions) {
+			if (delta >= this._filtredOptions.length) {
+				this._focusedOptionIndex = 0;
+			} else {
+				this._focusedOptionIndex = (this._focusedOptionIndex + delta + this._filtredOptions.length) % this._filtredOptions.length;
+				if (this._focusedOptionIndex < 0) {
+					this._focusedOptionIndex = this._filtredOptions.length - 1;
+				}
+			}
+			this.focusOption(this._focusedOptionIndex);
+		}
+	}
```

Comment:
Tiefe if/else-Verschachtelungen verschlechtern oft die Lesbarkeit von Funktionen. Meistens ist es am besten soweit wie möglich auf else zu verzichten und stattdessen mit return statements zu arbeiten, außer es geht nicht anders.

Als Beispiel: (Hab ich allerdings nicht direkt getest :))
```suggestion
private moveFokus(delta: number) {
	if (!this._filtredOptions) {
		return
	}
	let index = 0
	if (delta < this._filtredOptions.length) {
		const optionLength = this._filtredOptions.length
		index = (this._focusedOptionIndex + delta + optionLength) % optionLength;
		if (index < 0) {
			index = optionLength - 1;
		}
	}
	this._focusedOptionIndex = index
	this.focusOption(this._focusedOptionIndex);
}
```